xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -251,7 +269,7 @@ public PickResult pickSubchannel(PickSubchannelArgs args) {
       if (!enableOobLoadReport) {
         return PickResult.withSubchannel(subchannel,
             OrcaPerRequestUtil.getInstance().newOrcaClientStreamTracerFactory(
-                ((WrrSubchannel)subchannel).perRpcListener));
+                ((WrrSubchannel)subchannel).new OrcaReportListener(errorUtilizationPenalty)));
```

Comment:
Here the per-request listener can be created once at picker's construction time and reused.
A followup PR is fine.

Reply:
Done.
The per-request listener is created once (per channel & tracked with a map) in the picker's constructor.
I added a "fallback" case with `getOrDefault` for safety.